config/routes.rb

```diff
@@ -0,0 +1,17 @@
+Rails.application.routes.draw do
+
+  root "works#main"
+
+  resources :works
+
+  resources :users
+
+  resources :votes
+
+  get "/login", to: "sessions#new", as: "login"
+  post "/login", to: "sessions#create"
+  delete "/logout", to: "sessions#destroy", as: "logout"
+
+  get "/works/:id/upvote", to: "works#upvote", as: "upvote_work"
+  post "/works/:id/upvote", to: "works#upvote"
```

Comment:
It seems odd that you would have both a `get` and `post` request going to the same controller action - which one are you actually using?